scripts/perftune.py

```diff
@@ -171,19 +171,19 @@ def restart_irqbalance(banned_irqs):
     perftune_print("Restarting irqbalance: going to ban the following IRQ numbers: {} ...".format(", ".join(banned_irqs_list)))
 
     # Search for the original options line
-    opt_lines = list(filter(lambda line : re.search("^\s*{}".format(options_key), line), cfile_lines))
+    opt_lines = list(filter(lambda line : re.search(r"^\s*{}".format(options_key), line), cfile_lines))
     if not opt_lines:
         new_options = "{}=\"".format(options_key)
     elif len(opt_lines) == 1:
         # cut the last "
-        new_options = re.sub("\"\s*$", "", opt_lines[0].rstrip())
+        new_options = re.sub("\"\\s*$", "", opt_lines[0].rstrip())
```

Comment:
This is a bit confusing.
How about using `r"\"\s*$"`?

Reply:
This change looks wrong in general, @tchaikov.
Following the patch description what you want here is `r'"\s*$'`

Reply:
thank you Benny and Vlad for pointing this out. indeed, it should follow the patch description. just wanted to explain why it is right, not in the sense it matches with the description, but in the sense that it does the job:
```console
In [1]: import re

In [2]: line = 'hello"   '

In [3]: re.sub("\"\\s*$", "",  line)
Out[3]: 'hello'

In [4]: re.sub(r'"\s*$', "",  line)
Out[4]: 'hello'
```

will use `r'"\s*$'`, as single quotes makes the regex much simpler to read from developer's point of view.

Reply:
> just wanted to explain why it is right, not in the sense that it matches with the description, but in the sense that it does the job

That's strange but I confirm that your initial version gives the same match as the original (before the patch) one.
That's very weird because I was expecting that "\\s" regex would match a two char "\s" string (backslash + 's'). 

My apologies and I'm taking my "This change looks wrong in general" back.

Reply:
`\s` is special in at least python's re, i think. i am quoting CPython's document at https://docs.python.org/3/library/re.html#index-32

> \s
> For Unicode (str) patterns:
> Matches Unicode whitespace characters (which includes [ \t\n\r\f\v], and also many other characters, for example the non-breaking spaces mandated by typography rules in many languages).
> 
> Matches [ \t\n\r\f\v] if the [ASCII](https://docs.python.org/3/library/re.html#re.ASCII) flag is used.
> 
> For 8-bit (bytes) patterns:
> Matches characters considered whitespace in the ASCII character set; this is equivalent to [ \t\n\r\f\v].